src/neptune_tensorboard/integration/tensorflow_integration.py

```diff
@@ -13,255 +13,93 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from __future__ import unicode_literals
-
-import io
-import os
-import sys
-import time
+import warnings
+from functools import wraps
+from importlib.util import find_spec
 
 import tensorflow as tf
-from future.builtins import object
-from neptune.exceptions import NeptuneException
-from PIL import Image
+from neptune.types import File
 from pkg_resources import parse_version
 
-try:
-    from tensorflow_core.core.framework import summary_pb2
-except ImportError:
-    from tensorflow.core.framework import summary_pb2
+IS_GRAPHLIB_AVAILABLE = find_spec("tfgraphviz")
+if IS_GRAPHLIB_AVAILABLE:
+    import tfgraphviz as tfg
 
 _integrated_with_tensorflow = False
 
 
-def integrate_with_tensorflow(experiment_getter, prefix=False):
+def patch_tensorflow(run, base_namespace):
     global _integrated_with_tensorflow
 
     if _integrated_with_tensorflow:
```

Comment:
This could be simplified with:
```python
    if not _integrated_with_tensorflow:
        _integrate_with_tensorflow(run, base_namespace)
        _integrated_with_tensorflow = True
```